packages/amplify-graphql-rds-model-transformer/rds-lambda/clients/postgres/PostgresClient.ts

```diff
@@ -0,0 +1,67 @@
+import { Knex } from 'knex';
+import { DBClient } from '../DBClient.js';
+import { BaseRequest, Request } from '../../interfaces/BaseRequest.js';
+import { ListRequest } from '../../interfaces/ListRequest';
+
+export abstract class PostgresClient implements DBClient {
+    client: Knex;
+    abstract getClient(): Promise<any>;
+
+    executeRequest = async (request: BaseRequest): Promise<any> => {
+      const doExecute = async (): Promise<any> => {
+        switch (request.operation) {
+          case 'CREATE':
+            return this.executeCreate(request as Request);
+          case 'GET':
+            return this.executeGet(request as Request);
+          case 'UPDATE':
+            return this.executeUpdate(request as Request);
+          case 'LIST':
+            return this.executeList(request as ListRequest);
+          case 'DELETE':
+            return this.executeDelete(request as Request);
+          default:
+            throw Error('Invalid operation');
+        }
+      };
+
+      const result = await doExecute();
+      const data = {};
+      data[request.operationName] = result;
+      return result;
+    }
+
+    private executeCreate = async (request: Request): Promise<any> => {
+      const result = await (await this.getClient())(request.table).insert(request.args.input).returning('*');
+      return result;
+    }
+
+    private executeGet = async (request: Request): Promise<any> => {
+      const query = (await this.getClient())(request.table).where('id', request.args.input.get('id')).select();
```

Comment:
**Throughout the client:** The hard-coded field `id` is not guaranteed to exist in all tables right? Or is there any `Knex` specific detail that I am missing here?

Reply:
This is just something I've left in for the moment - I'm going to be continuing to improve this over the next few PRs and CPK support is part of that